frontend/public/components/overview/namespace-overview.tsx

```diff
@@ -41,9 +46,70 @@ const OverviewNamespaceSummary = ({ns}) => <div className="group">
   </div>
 </div>;
 
+const QuotaBox = ({quota, resourceTypes}) => {
+  const scopes = _.get(quota, ['spec', 'scopes']);
+  return <div className="col-xs-6 quota-dashboard-column">
+    <div className="group group__body full-width-and-height" key={quota.metadata.name}>
+      <div className="group__title group__title--no-side-borders">
+        <h2 className="h3 full-width-and-height">
+          <ResourceLink kind="ResourceQuota" name={quota.metadata.name} className="co-resource-link-truncate"
+            namespace={quota.metadata.namespace} title={quota.metadata.name} />
+        </h2>
+      </div>
+      <div className="container-fluid group__graphs">
+        <QuotaGaugeCharts quota={quota} resourceTypes={resourceTypes} rowClass="co-resource-quota-chart-row--compact"
+          chartClass="graph-wrapper--compact" />
+        {scopes && <dl className="co-m-pane__details quota-dashboard-scopes">
+          <QuotaScopes scopes={scopes} compact={true} />
+        </dl>}
+      </div>
+    </div>
+  </div>;
+};
+
+class QuotaBoxes extends React.PureComponent<QuotaBoxesProps> {
```

Comment:
Any reason not to use a stateless functional component? Firehose does implement `shouldComponentUpdate`. (Was pure component a deliberate choice?)

Reply:
no reason, ok, I'll switch to stateless component

Reply:
fixed